### PR TITLE
Unit test selection method

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves: #10053
resolves: https://github.com/dbt-labs/dbt-core/issues/9895

### Problem

unit tests could not be passed in using --select `unit_test:*`

### Solution

implement unit test selection method!

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

Some additional 🎩 
```sh
❯ dbt ls --resource-type unit_test
20:41:11  Running with dbt=1.9.0-a1
20:41:11  target not specified in profile 'postgres', using 'default'
20:41:11  Registered adapter: postgres=1.9.0-a1
20:41:12  Found 15 models, 6 seeds, 20 data tests, 6 sources, 13 metrics, 684 macros, 6 semantic models, 4 unit tests
unit_test:jaffle_shop.test_supply_costs_sum_correctly
unit_test:jaffle_shop.test_order_items_compute_to_bools_correctly
unit_test:jaffle_shop.test_recursive_cte
unit_test:jaffle_shop.test_does_location_opened_at_trunc_to_date

❯ dbt ls --select unit_test:jaffle_shop.test_recursive_cte
20:41:18  Running with dbt=1.9.0-a1
20:41:18  target not specified in profile 'postgres', using 'default'
20:41:19  Registered adapter: postgres=1.9.0-a1
20:41:19  Found 15 models, 6 seeds, 20 data tests, 6 sources, 13 metrics, 684 macros, 6 semantic models, 4 unit tests
unit_test:jaffle_shop.test_recursive_cte

❯ dbt test --select unit_test:jaffle_shop.test_recursive_cte
20:41:24  Running with dbt=1.9.0-a1
20:41:24  target not specified in profile 'postgres', using 'default'
20:41:25  Registered adapter: postgres=1.9.0-a1
20:41:25  Found 15 models, 6 seeds, 20 data tests, 6 sources, 13 metrics, 684 macros, 6 semantic models, 4 unit tests
20:41:25  
20:41:26  Concurrency: 1 threads (target='default')
20:41:26  
20:41:26  1 of 1 START unit_test recursive_cte::test_recursive_cte ....................... [RUN]
20:41:26  1 of 1 PASS recursive_cte::test_recursive_cte .................................. [PASS in 0.14s]
20:41:26  
20:41:26  Finished running 1 unit test in 0 hours 0 minutes and 1.27 seconds (1.27s).
20:41:27  
20:41:27  Completed successfully
20:41:27  
20:41:27  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

### CODE REVIEW

**Review by gshank — APPROVED** [review]

**Review by peterallenwebb — APPROVED** [review]

File: tests/unit/test_graph_selector_methods.py [hunk 1]
```diff
@@ -586,6 +588,24 @@ def test_select_saved_query_by_tag(manifest: Manifest) -> None:
     search_manifest_using_method(manifest, method, "any_tag")
 
 
+def test_select_unit_test(manifest: Manifest) -> None:
+    test_model = make_model("test", "my_model", "select 1 as id")
+    unit_test = make_unit_test("test", "my_unit_test", test_model)
+    manifest.unit_tests[unit_test.unique_id] = unit_test
```
Comment (ChenyuLInx) [inline tests/unit/test_graph_selector_methods.py:594]: Do you feel like this is easier to use or we somehow have a fixture called `unittests` that will be required by manifest and you return all unittest nodes in it easier? Or maybe not so much difference?
cc @QMalcolm

**Review by ChenyuLInx — APPROVED** [review]

**Review by ChenyuLInx — COMMENTED** [review]
Ideally the `test_graph_selectore_methods.py` should be moved to
`tests/unit/graph/test_selector_methods.py`, but I can do it in my big reorganizing everything issue

File: tests/unit/test_graph_selector_methods.py (hunk 1 above)
Comment (MichelleArk) [reply tests/unit/test_graph_selector_methods.py:594]: I thought this was very easy to use! and keeps unit tests more narrow and isolated from one another in their setup

**Comment by MichelleArk** [comment]
@ChenyuLInx -- I'll do the move here! Should be quick :)

**Pull request merged**